### ISSUE TITLE
Incorrect coin total

### ISSUE THREAD

**Issue by allegro16** [PaycoinFoundation/paycoin#266]
Hi,

I run my Paycoin wallet 24/7. A few days ago i noticed something weird: it suddenly started staking like crazy, like every 10m or so. With only 4000 (399x something) XPY in the wallet. Some orphans, some legit.

Then: my coin total suddenly rose to 5500, so up 1500XPY. I thought: maybe a staking error?
 So, shutdown the wallet, then the wallet showed 399X briefly, before jumping to 5500 again.
 Then, after i opened the wallet for minting, it jumped up to 9300XPY again.. Tried again.. Jumped to 13900 or so.

I was able to fix the jumping when minting with a repair wallet.. But the coin total kept the fixed 13900 ( i have a clean backup though)..
 The coins CAN'T be spend though, i get a double spend warning.

I AM able to send the incorrect amount to one of the addresses in the wallet though. Still CAN'T spend it..

But a repairwallet now ADDS the original 399X (or the available not minting amount) to the incorrect value.
 Still can't spend any but it DOES increase the staking rate from once in a few days to like 6 an hour. About half of them are valid! 

Not sure where to send the bug to though..

I'm happy to send any log/proof or something if needed.

Is there a bugreport reward? :) Joking, but i would appreciate.
Do note: i imported the Micro Prime wallet key, maybe that's of importance?

Allegro16

![bug1](https://cloud.githubusercontent.com/assets/15080463/10419782/3e3d0f5a-7082-11e5-9058-4ebb9c5005e1.png)
![bug2](https://cloud.githubusercontent.com/assets/15080463/10419779/3e3bb024-7082-11e5-9774-a0e4ae390001.png)
![bug3](https://cloud.githubusercontent.com/assets/15080463/10419780/3e3c016e-7082-11e5-9092-8c740de7ce84.png)
![bug4](https://cloud.githubusercontent.com/assets/15080463/10419778/3e3b4814-7082-11e5-9287-a1e4cbf87906.png)
![bug5](https://cloud.githubusercontent.com/assets/15080463/10419781/3e3cac5e-7082-11e5-944f-574d2f8c7800.png)
![repair](https://cloud.githubusercontent.com/assets/15080463/10419777/3e3899fc-7082-11e5-84eb-5086382fe842.png)
![after repair](https://cloud.githubusercontent.com/assets/15080463/10419813/2977064c-7083-11e5-907e-69f20371158b.png)

**Comment by allegro16** [comment]
It seems a 0.3.1.1 bug, 0.3.1.0 should be fine..

I'll have to try sending coin from the wallet i guess, as after redownloading the wallet i have 1678XPY left of my 3998XPY before the bug happened. It's not adding XPY now, but eating..
I still have several backups i will try once the blockchain is in completely.

**Comment by allegro16** [comment]
P.S: https://www.paycointalk.org/topic/672/paycoin-wallet-0-3-1-1-bug/ for more on the issue..

**Comment by mitchellcash** [comment]
I think it's v0.3.1.0 as well.

I feel it's something to do with Primes and some of the dates set around them in the code but not sure yet.

**Comment by allegro16** [comment]
Could be, but if the wallet can't be fixed, i think i'm out. I hope i didn't actually lose any coin..
This is the weirdest bug in any wallet i encountered.

I did add my Micro Prime key though, maybe that's the issue?

**Comment by mitchellcash** [comment]
No coins should be lost. Once the bug is resolved the worst will be that you need to redownload the block chain but all existing coins should remain.

**Comment by allegro16** [comment]
Yes, i noticed. It was due too it was at a point in the blockchain where it was staking more than half of my coins and it wasn't visible yet. Now it's 155 blocks left (takes quite a while), but it seems it should be more or less OK. There could still be a 3XPY difference, as the backup is made after i noticed the issue.

Anyway, a quite serious bug as it messes the wallet up quite a bit. In my case:
the sudden rise from about 4000XPY to 5500, then 9713, 13706 and higher.. The increased coins from stakes..)

**Comment by allegro16** [comment]
Sigh.. Had a relatively clean backup.. That was updated with the blockchain.. Then out of sync.. Then somehow more than doubled again.. Now trying 0.3.1.0 again..

![bug](https://cloud.githubusercontent.com/assets/15080463/10426899/e5bbd998-70e4-11e5-8bad-7a678dad4dd3.png)

**Comment by IngCr3at1on** [comment]
As for the actual balance issue, etc. You're not the only one reporting an issue though so far as I can tell it's not code related.

To answer the question about adding your microprime address, taht would make no difference at all.

**Comment by allegro16** [comment]
Well, if it's not code related.. How would this happen then? In 2/3 years running wallets, it's the only wallet having such a serious issue. Indeed if you look at the Paycointalk thread: i'm not alone.
That doesn't help any of us though, as the proposed temporary fixes there don't seem to work either.

**Comment by IngCr3at1on** [comment]
A fork can occur without any code issue simply by a few nodes passing blocks that others consider duplicates by the time they got them (normally these would be orphaned as no other nodes would accept them but in this instance at least 1 or 2 nodes did accept them).

Unfortunately it's more probable with POS than with POW which is why you don't often see such behavior on Bitcoin (generally). The solution is to resync from the correct chain but the problem is there are still nodes with the invalid chain so simply deleting the blockchain data and resyncing isn't a guaranteed solution, instead one needs to addnode known good nodes (sadly I don't have an IP I can provide for that), as more nodes get corrected the odds of syncing from the correct chain improve.

Basically it's Murphy's Law in action, there are quite a few variables in play; block drift being one major factor which is particularly higher than it should be (this was done by the original devs), lowering this however is something that has to be done carefully or we could lock up the chain entirely (which IMHO is worse than a fork). Soon as I have a node that I can provide an IP address for I'll let you know, in the meantime your options are to wait or try deleting the blockchain data and attempting to resync again.

**Comment by allegro16** [comment]
Also: i deleted ALL files in the wallet folder, not only the blockchain, so it's creating fresh files for everything. (i did backup), so nothing old can mess with the wallet. 
I will stay on 0.3.1.0 for now..

**Comment by IngCr3at1on** [comment]
Technically speaking there's nothing to stop the 0.3.1.0 wallet from having the same issue however if you find it works for you there's certainly no reason not to use it either. Sorry again for the issue you are having.

Might sound crazy but I wish it was something tangible in the code least then it would be something I could fix at the code level ;)

**Comment by allegro16** [comment]
Well, it doesn't work YET.. AS  have 2 wallets currently struggling to update..
But if i get 1 (the old wallet) or the new with imported keys to stay synced, i'll use that version for now. As the issues started with 0.3.1.1 in my case..

**Comment by allegro16** [comment]
Just FYI:

I downgraded to 0.3.1.0. The data was still out of sync most of the time..
I did a checkwallet and a repair wallet, this time it correctly fixed the data (in ALL wallets, not only the 3998XPY one, but the 9700, 15000XPY as well) WITHOUT adding the difference on top.
It seems there were some orphans in the 3998XPY total already, it got fixed to 3987.790923, which corresponds exactly to the blockchain..

So it IS a problem with 0.3.1.1 especially, as the problem arose under 0.3.1.1 (the sudden +1500XPY and staking 6x an hour) and the repairwallet under 0.3.1.1 made things worse. 0.3.1.0 actually fixed them..

I'll wait for v0.3.1.2 or later before updating again.

I advise everyone to do the same as it's not that easy to repair sometimes.

Unfortunate for the 11000XPY i suddenly gained.. :( I could have used them.
Hopefully the Micro Primes will work soon and i will be able to stake most of my XPY.. (I guess not, was based on 100 Prime Stakers..)..

**Comment by IngCr3at1on** [comment]
The reason repairwallet didn't work previously was because your chain data was wrong, as for the xpy that you gained yes those were blocks staked on a fork so they don't exist on the main/proper blockchain.

Not to be argumentative but so far as I can tell there's still nothing that tells me this is specifically a 0.3.1.1 issue but I am glad you're back on track (still waiting on my local node to sync to play with as my remote node is working fine).

**Comment by allegro16** [comment]
Well, downgrading does seem to fix the issues for most. So, it at least looks there is something wrong with 0.3.1.1..
Hopefully it will be sorted soon.

Not happy i lost about 11000 coins (which were illegitimate, i know..), but happy i posted it here and on PayCointalk and it gets fixed.

**Comment by allegro16** [comment]
It might be an idea to offer some kind of reward for finding and reporting bugs as i did? This usually speeds up debugging, etc. LIke some XPY or like a Micro Prime upgrade?

**Comment by allegro16** [comment]
It still stakes too much but it no longer corrupts the wallet. I'm missing about 6-10XPY an hour than are orphaned.. About 150XPY a day or more. FIX please or keep us updated..

**Comment by PaycoinMaster** [comment]
Hello @allegro16, The Core Dev Team is looking into the issue and so is every available person on the network. We are working hard to figure out what caused this issue overall and get the coin back on track.

**Comment by allegro16** [comment]
Thanks, we'll await your news. 
And a thanks would be appreciated. :) (As you have no reward program for finding bugs)

**Comment by HyperCommNet** [comment]
I agree there should be a reward program. Issue is who is going to fund it. I say we launch a committee to discuss the possibilities of the said bug reward program.

**Comment by HyperCommNet** [comment]
Who wants to be in this committee?

**Comment by IngCr3at1on** [comment]
Old issue related to fork, closing.
